batavia/VirtualMachine.js

```diff
@@ -1300,15 +1300,26 @@ VirtualMachine.prototype.byte_COMPARE_OP = function(opnum) {
 VirtualMachine.prototype.byte_LOAD_ATTR = function(attr) {
     var obj = this.pop();
     var val;
-    if (obj.__getattr__ === undefined) {
+
+    if (obj.__getattribute__ === undefined) {
         val = obj[attr];
         if (val === undefined) {
-            throw new builtins.AttributeError(
-                "'" + obj.__class__.__name__ + "' object has no attribute '" + attr + "'"
-            );
+            if (obj.__getattr__ === undefined) {
+                throw new builtins.AttributeError(
+                    "'" + obj.__class__.__name__ + "' object has no attribute '" + attr + "'"
+                );
+            } else if (obj.__getattr__ instanceof Function) {
+                val = obj.__getattr__(attr);
+            } else if (obj.__getattr__ instanceof types.Function) { // Shouldn't these already be methods?
```

Comment:
No - the Function is only bound to the instance as part of the __getattr__ process. The class has a single, common `types.Function` instance; each object instance gets a separate types.Method instance. In theory, this *could* be cached after first access, or at time of object construction; but Method is such a lightweight class it doesn't really matter.